pkg/reconciler/buildrun/resources/sources/local_copy.go

```diff
@@ -0,0 +1,25 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package sources
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/shipwright-io/build/pkg/config"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+// AppendLocalCopyStep defines and append a new task based on the waiter container template, passed
+// by the configuration instance.
+func AppendLocalCopyStep(cfg *config.Config, taskSpec *tektonv1beta1.TaskSpec, timeout *metav1.Duration) {
+	step := tektonv1beta1.Step{Container: *cfg.WaiterContainerTemplate.DeepCopy()}
+	step.Name = "local-copy"
```

Comment:
For consistency, the pattern for step names for sources is `source-${sourceName}`, the sourceName comes from the BuildSource object. See https://github.com/shipwright-io/build/blob/v0.7.0/pkg/reconciler/buildrun/resources/sources/git.go#L48 or https://github.com/shipwright-io/build/blob/v0.7.0/pkg/reconciler/buildrun/resources/sources/bundle.go#L38 (I know that we atm violate this with sources-http for HTTP sources as we still combine them all into a single step).

Reply:
Here, the CLI watches for pods with the [expected label selector](https://github.com/shipwright-io/cli/blob/b5eb161f13dc2ed6b49d87e5c9b64147ca18233a/pkg/shp/cmd/build/upload.go#L271-L278), and once found, the CLI [waits for running](https://github.com/shipwright-io/cli/blob/b5eb161f13dc2ed6b49d87e5c9b64147ca18233a/pkg/shp/cmd/build/upload.go#L234) state before trying to upload. Then, the [upload is target to a static container name](https://github.com/shipwright-io/cli/blob/b5eb161f13dc2ed6b49d87e5c9b64147ca18233a/pkg/shp/cmd/build/upload.go#L237-L242).

Given the nature of having a specific container to upload (waiter), having a static container name is straightforward from the CLI perspective. With that said, could we then document why this specific container name has a particular name, and also, we should export this name as a constant that can be imported on the CLI.

What do you think?

Reply:
Yep, a static container name makes sense. I'd still recommend for consistency that it is something like `source-local`.